<launch>
<!-- ============ Arguments ============ -->
  <!-- General -->
  <arg name="namespace" default="data"/> 
  
  <!-- Flat Dataset args -->
  <arg name="bag_name" default="/home/lukas/Documents/Datasets/flat_dataset/run1.bag"/> 
  <arg name="bag_play_rate" default="1"/>  <!-- Realtime factor -->
  <arg name="use_sim_time" default="false"/>
  <arg name="use_detectron" default="false"/>
  
  <!-- Mapper -->
  <arg name="config" default="demo"/>



<!-- ============ Run ============ -->
  <param name="/use_sim_time" value="$(arg use_sim_time)" />
  
  <!-- Play data -->
  <include file="$(find panoptic_mapping_utils)/launch/play_flat_dataset.launch" pass_all_args="true" />
  
  <!-- Mapper -->
  <node name="panoptic_mapper" pkg="panoptic_mapping_ros" type="panoptic_mapper_node" output="screen">
    <!-- Config -->
    <rosparam file="$(find panoptic_mapping_ros)/config/mapper/$(arg config).yaml"/>

    <!-- Input -->
    <remap from="color_image_in" to="$(arg namespace)/color_image"/>
    <remap from="depth_image_in" to="$(arg namespace)/depth_image"/>
    <remap from="segmentation_image_in" to="$(arg namespace)/segmentation_image"/>
    <remap from="labels_in" to="$(arg namespace)/segmentation_labels"/>
  </node>  
  
  <!-- RVIZ Visualization-->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find panoptic_mapping_ros)/config/rviz/demo.rviz" output="screen"/>

  <!-- Path visualization -->
  <node name="path_visualizer" pkg="panoptic_mapping_utils" type="path_visualizer.py" output="screen">
    <remap from="~pose_in" to="$(arg namespace)/pose" />
    <param name="use_arrow" value="false" /> 
    <param name="length" value="0.5" />
  </node>

</launch>
